src/Sylius/Bundle/CoreBundle/CatalogPromotion/Announcer/CatalogPromotionAnnouncer.php

```diff
@@ -14,19 +14,19 @@
 namespace Sylius\Bundle\CoreBundle\CatalogPromotion\Announcer;
 
 use Sylius\Bundle\CoreBundle\Calculator\DelayStampCalculatorInterface;
-use Sylius\Calendar\Provider\DateTimeProviderInterface;
 use Sylius\Component\Core\Model\CatalogPromotionInterface;
 use Sylius\Component\Promotion\Event\CatalogPromotionCreated;
 use Sylius\Component\Promotion\Event\CatalogPromotionEnded;
 use Sylius\Component\Promotion\Event\CatalogPromotionUpdated;
+use Symfony\Component\Clock\ClockInterface;
 use Symfony\Component\Messenger\MessageBusInterface;
 
 final class CatalogPromotionAnnouncer implements CatalogPromotionAnnouncerInterface
 {
     public function __construct(
         private MessageBusInterface $eventBus,
         private DelayStampCalculatorInterface $delayStampCalculator,
-        private DateTimeProviderInterface $dateTimeProvider,
```

Comment:
All of these changes should be noted in the upgrade file, even more so since there was no deprecation/transition between the two, it just poofed.

Reply:
Ah right, I'll open also a PR against 1.13 adding deprecations.